modules/farm/farm_livestock/farm_livestock.install

```diff
@@ -508,3 +508,54 @@ function farm_livestock_update_7007(&$sandbox) {
     module_enable(array($module));
   }
 }
+
+/**
+ * Remove duplicate area references from movement quickform logs.
+ */
+function farm_livestock_update_7008(&$sandbox) {
+
+  // Setup. Find observation logs associated with the farm_livestock_move_form
+  // that need to be updated.
+  if (!isset($sandbox['progress'])) {
+    $sandbox['progress'] = 0;
+
+    // Query observation logs linked to this quick form.
+    $query = db_select('farm_quick_entity', 'fqe');
+    $query->addField('fqe', 'entity_id');
+    $query->condition('fqe.entity_type', 'log');
+    $query->condition('fqe.quick_form_id', 'farm_livestock_move_form');
+    $log_alias = $query->join('log', 'l', 'fqe.entity_id = l.id');
+    $query->condition($log_alias . '.type', 'farm_observation');
+    $log_ids = $query->execute()->fetchCol();
+
+    // Finish the update if there are no logs to update.
+    if (empty($log_ids)) {
+      $sandbox['#finished'] = 1;
+      return;
+    }
+
+    $sandbox['log_ids'] = $log_ids;
+    $sandbox['total'] = count($log_ids);
+  }
+
+  // Load the Nth log we need to process.
+  $log = log_load($sandbox['log_ids'][$sandbox['progress']]);
+
+  // Load areas that each log references.
+  $log_wrapper = entity_metadata_wrapper('log', $log);
+  $areas = $log_wrapper->field_farm_area->value();
+
+  // Only update areas if some exist.
+  if (!empty($areas)) {
+
+    // Get the unique areas.
+    $new_areas = array_unique($areas, SORT_REGULAR);
```

Comment:
I wonder if we should add another condition here to ensure that we are only saving the log if it indeed has/had duplicates. As it stands now, the `if (!empty($areas))` condition alone will most likely match all logs.

Reply:
Right now this matches all logs *with areas*. But yes, it will save any log that had at least one area. It seems pretty safe to me, but we could add another condition. Maybe:

```php
if (!empty($areas) && count($areas) > 1) {

  // Get the unique areas.
  $new_areas = array_unique($areas, SORT_REGULAR);

  if (count($areas) > count($new_areas)) {
    // update & save log
  }
}
```

Reply:
Yea! I was thinking something similar. That would be simple enough...

It's a minor optimization, I agree. But easy enough to include. The extra benefit we get (still very minor) is that the `changed` timestamp will only be updated on logs that needed to be changed... so if there's an issue it's another metric we can look at to try to fix things.